devito/propagator.py

```diff
@@ -30,10 +33,20 @@ def __init__(self, name, nt, shape, spc_border=0, forward=True, time_order=0):
         # Start with the assumption that the propagator needs to save the field in memory at every time step
         self._save = True
         # This might be changed later when parameters are being set
-
+        self.profile = os.environ.get(self._ENV_VAR_PROFILE) == "1"
```

Comment:
I don't think enabling profiling via an env flag is a great idea. I would instead propose to enable this via a Python option that defaults to `True` so that we run with profiling by default.